src/lib.rs

```diff
@@ -0,0 +1,198 @@
+//! # Substrate
+//!
+//! Substrate is a Rust framework for building blockchains in a modular and extensible way. While in
+//! itself un-opinionated, it is the main engine behind the Polkadot ecosystem.
+//!
+//! [![github]](https://github.com/paritytech/substrate/) - [![polkadot]](https://polkadot.network)
+//!
+//! This crate in itself does not contain any code and is just meant ot be a documentation hub for
+//! substrate-based crates.
+//!
+//! ## Overview
+//!
+//! Substrate approaches blockchain development with a focus on **safety**, **correctness**, and
+//! **upgradeability**.
```

Comment:
I think its generality is also highly interesting. Substrate is using generic types quite often to make it possible to adapt to any use case.